### ISSUE TITLE
Provide source code tarballs for future stable releases

### ISSUE THREAD

**Issue by ilyazzz** [Chatterino/chatterino2#2566]
Currently, for stable chatterino versions only pre-built binaries (exe/dmg/appimage) are provided. 

If one wants to build a stable version of chatterino from source, they run into certain issues:

- The source archives that get auto-generated by github don't include submodules
~~- Building from a specific tag/commit fails as the submodules still track the latest version, and have been updated to be incompatible since the chatterino tag.~~ This is wrong, it seems to be an issue with how flatpak manifests update submodules

More specifically, I am trying to get chatterino [packaged into flatpak](https://github.com/flathub/flathub/pull/2175) and they require reproducible builds, which doesn't happen when either the repo or it's submodules are building from git master.

I have looked at how [fedora does it's chatterino package](https://src.fedoraproject.org/rpms/chatterino2/blob/rawhide/f/chatterino2.spec), and it locks all of the submodules at a specific commit. IMO this is not a good solution, and it would get very tedious to update each submodule invidiually while checking for it's compatibility (perhaps this is also why the fedora package is currently broken, #2565 ).

A solution for all of this would be to provide a source code tarball with all of the submodules included.

Also on the topic of packaging, the [appdata.xml file](https://github.com/Chatterino/chatterino2/blob/master/resources/com.chatterino.chatterino.appdata.xml) doesn't fully adhere to the standards:

```
$ appstream-util validate resources/com.chatterino.chatterino.appdata.xml 

resources/com.chatterino.chatterino.appdata.xml: FAILED:
• attribute-invalid     : <screenshot> width too large [https://chatterino.com/img/screenshot-3.png] maximum is 1600px
• attribute-invalid     : <screenshot> height too large [https://chatterino.com/img/screenshot-3.png] maximum is 900px
• tag-missing           : <release> required
Validation of files failed
```

**Comment by pajlada** [comment]
I'm fine with providing source code tarballs as part of the nightly build process

**Comment by ilyazzz** [comment]
For the purposes of packaging this is "resolved" for me, since the 2.3 tag works properly. It looks like there was an issue specific to the 2.2.3 tag.
